aptos-move/move-examples/bonding_curve_launchpad/sources/liquidity_pair.move

```diff
@@ -0,0 +1,277 @@
+        liquidity_pair.apt_reserves = apt_updated_reserves;
+
+        event::emit(LiquidityPairReservesUpdated {
+            old_fa_reserves: old_fa_reserves,
+            old_apt_reserves: old_apt_reserves,
+            new_fa_reserves: fa_updated_reserves,
+            new_apt_reserves: apt_updated_reserves
+        });
+        event::emit(LiquidityPairSwap {
+            is_fa_else_apt: false,
+            gained: (apt_gained as u128),
+            swapper_address: swapper_address
+        });
+    }
+
+    public(friend) fun internal_swap_apt_to_fa(transfer_ref: &fungible_asset::TransferRef, account: &signer,  fa_object_metadata: Object<Metadata>, amountIn: u64) acquires LiquidityPairSmartTable {
+        let liquidity_pair_smartTable = borrow_global_mut<LiquidityPairSmartTable>(@resource_account);
+        assert!(smart_table::contains(&mut liquidity_pair_smartTable.liquidity_pairs, fa_object_metadata), ELIQUIDITY_PAIR_DOES_NOT_EXIST);
+        let liquidity_pair = smart_table::borrow_mut(&mut liquidity_pair_smartTable.liquidity_pairs, fa_object_metadata);
+        assert!(liquidity_pair.is_enabled, ELIQUIDITY_PAIR_DISABLED);
+
+        let swapper_address = signer::address_of(account);
+        let (fa_gained, apt_given, fa_updated_reserves, apt_updated_reserves) = get_amount_out(
+            liquidity_pair.fa_reserves,
+            liquidity_pair.apt_reserves,
+            false,
+            amountIn
+        );
+        let does_primary_store_exist_for_swapper = primary_fungible_store::primary_store_exists(swapper_address, fa_object_metadata);
+        if(!does_primary_store_exist_for_swapper){
+            primary_fungible_store::create_primary_store(swapper_address, fa_object_metadata);
+        };
+        aptos_account::transfer(account, @resource_account, apt_given);
+        primary_fungible_store::transfer_with_ref(transfer_ref, @resource_account, swapper_address, fa_gained);
+        let old_fa_reserves = liquidity_pair.fa_reserves;
+        let old_apt_reserves = liquidity_pair.apt_reserves;
+        liquidity_pair.fa_reserves = fa_updated_reserves;
+        liquidity_pair.apt_reserves = apt_updated_reserves;
+
+        event::emit(LiquidityPairReservesUpdated {
+            old_fa_reserves: old_fa_reserves,
+            old_apt_reserves: old_apt_reserves,
+            new_fa_reserves: fa_updated_reserves,
+            new_apt_reserves: apt_updated_reserves
+        });
+        event::emit(LiquidityPairSwap {
+            is_fa_else_apt: true,
+            gained: (fa_gained as u128),
+            swapper_address: swapper_address
+        });
+
+        if(apt_updated_reserves > APT_LIQUIDITY_THRESHOLD && liquidity_pair.is_enabled){
+            // Disable Bonding Curve Launchpad pair.
+            liquidity_pair.is_enabled = false;
+            liquidity_pair.is_frozen = false;
+            // Offload onto third party, public DEX.
+            router::create_pool_coin<AptosCoin>(fa_object_metadata, false);
```

Comment:
Wouldn't this try to create the pool twice if this is called a second time? Should it only the create the pool once?

Reply:
`liquidity_pair.is_enabled` is immediately disabled, so the graduation conditional can only be entered once.